server/protocols/jwt/src/main/java/org/apache/james/jwt/OidcJwtTokenVerifier.java

```diff
@@ -65,17 +63,19 @@ public static <T> Optional<T> getClaimWithoutSignatureVerification(String token,
         }
     }
 
-    public static Publisher<String> verifyWithMaybeIntrospection(String jwtToken, URL jwksURL, String claimName, Optional<IntrospectionEndpoint> introspectionEndpoint) {
+    public static Publisher<String> verifyWithIntrospection(String jwtToken, URL jwksURL, String claimName, IntrospectionEndpoint introspectionEndpoint) {
         return Mono.fromCallable(() -> verifySignatureAndExtractClaim(jwtToken, jwksURL, claimName))
             .flatMap(optional -> optional.map(Mono::just).orElseGet(Mono::empty))
-            .flatMap(claimResult -> {
-                if (introspectionEndpoint.isEmpty()) {
-                    return Mono.just(claimResult);
-                }
-                return Mono.justOrEmpty(introspectionEndpoint)
-                    .flatMap(endpoint -> Mono.from(INTROSPECTION_CLIENT.introspect(endpoint, jwtToken)))
-                    .filter(TokenIntrospectionResponse::active)
-                    .map(activeToken -> claimResult);
-            });
+            .flatMap(claimResult -> Mono.from(CHECK_TOKEN_CLIENT.introspect(introspectionEndpoint, jwtToken))
+                .filter(TokenIntrospectionResponse::active)
+                .map(activeResponse -> claimResult));
+    }
+
+    public static Publisher<String> verifyWithUserinfo(String jwtToken, URL jwksURL, String claimName, URL userinfoEndpoint) {
+        return Mono.fromCallable(() -> verifySignatureAndExtractClaim(jwtToken, jwksURL, claimName))
+            .flatMap(optional -> optional.map(Mono::just).orElseGet(Mono::empty))
+            .flatMap(claimResult -> Mono.from(CHECK_TOKEN_CLIENT.useInfo(userinfoEndpoint, jwtToken))
+                // .filter(userinfoResponse -> userinfoResponse.getPreferredUsername().equals(claimResult))
```

Comment:
Not sure here
If we want to check the `preferred_username` must be equal to `claimResult`, we need to add 1 more configure. 
Eg: `auth.oidc.userinfo.claimname`

Reply:
Why not use `imapservers.imapserver.auth.oidc.claim` ?

Reply:
The first I see the 2 payloads are different, So a bit worried. 
But I rethinking again, It's ok. We can use `imapservers.imapserver.auth.oidc.claim`